lib/segment/src/index/query_optimization/condition_converter.rs

```diff
@@ -274,9 +274,7 @@ pub fn get_match_checkers(index: &FieldIndex, cond_match: Match) -> Option<Condi
             FieldIndex::FullTextIndex(full_text_index) => {
                 let parsed_query = full_text_index.parse_query(&text);
                 Some(Box::new(move |point_id: PointOffsetType| {
-                    full_text_index
-                        .get_doc(point_id)
-                        .map_or(false, |doc| parsed_query.check_match(doc))
+                    full_text_index.check_match(&parsed_query, point_id)
```

Comment:
Here is an affected point. Previously we checked the match between the two documents. Now, we use the postings list to do this check. It is less efficient, here is random access in the posting list for each token